test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -149,209 +158,50 @@
-                injector.getInstance(CircuitBreakerService.class), mapperService);
-        bitsetFilterCache = new BitsetFilterCache(idxSettings, new BitsetFilterCache.Listener() {
-            @Override
-            public void onCache(ShardId shardId, Accountable accountable) {
-
-            }
-
-            @Override
-            public void onRemoval(ShardId shardId, Accountable accountable) {
+    protected Collection<Class<? extends Plugin>> getPlugins() {
+        return Collections.emptyList();
+    }
 
-            }
-        });
-        indicesQueriesRegistry = injector.getInstance(IndicesQueriesRegistry.class);
-        //create some random type with some default field, those types will stick around for all of the subclasses
-        currentTypes = new String[randomIntBetween(0, 5)];
-        for (int i = 0; i < currentTypes.length; i++) {
-            String type = randomAsciiOfLengthBetween(1, 10);
-            mapperService.merge(type, new CompressedXContent(PutMappingRequest.buildFromSimplifiedDef(type,
-                    STRING_FIELD_NAME, "type=text",
-                    STRING_FIELD_NAME_2, "type=keyword",
-                    INT_FIELD_NAME, "type=integer",
-                    DOUBLE_FIELD_NAME, "type=double",
-                    BOOLEAN_FIELD_NAME, "type=boolean",
-                    DATE_FIELD_NAME, "type=date",
-                    OBJECT_FIELD_NAME, "type=object",
-                    GEO_POINT_FIELD_NAME, GEO_POINT_FIELD_MAPPING,
-                    GEO_SHAPE_FIELD_NAME, "type=geo_shape"
-            ).string()), MapperService.MergeReason.MAPPING_UPDATE, false);
-            // also add mappings for two inner field in the object field
-            mapperService.merge(type, new CompressedXContent("{\"properties\":{\"" + OBJECT_FIELD_NAME + "\":{\"type\":\"object\","
-                            + "\"properties\":{\"" + DATE_FIELD_NAME + "\":{\"type\":\"date\"},\"" + INT_FIELD_NAME + "\":{\"type\":\"integer\"}}}}}"),
-                    MapperService.MergeReason.MAPPING_UPDATE, false);
-            currentTypes[i] = type;
-        }
-        namedWriteableRegistry = injector.getInstance(NamedWriteableRegistry.class);
+    protected void initializeAdditionalMappings(MapperService mapperService) throws IOException {
     }
 
     @AfterClass
     public static void afterClass() throws Exception {
-        injector.getInstance(ClusterService.class).close();
-        terminate(injector.getInstance(ThreadPool.class));
-        injector = null;
-        index = null;
-        currentTypes = null;
-        namedWriteableRegistry = null;
-        randomTypes = null;
-        indicesQueriesRegistry = null;
-        indexFieldDataService = null;
-        searchModule = null;
-        idxSettings = null;
-        similarityService = null;
-        mapperService = null;
-        bitsetFilterCache = null;
-        scriptService = null;
+        holder.close();
```

Comment:
this will cause an NPE if you fail to initialize maybe use `IOUtils.close(holder)`